docs/products/opensearch/howto/restore-snapshot-opensearch.md

```diff
@@ -0,0 +1,279 @@
+  Elasticsearch 7.10.2 or below are supported).
+- The feature flag enabled for your project, allowing external
+  migrations to be configured and executed.
+- Access credentials for the cloud provider hosting the snapshot.
+
+### Gather required parameters
+
+For each cloud provider, ensure you have the following information:
+
+- **Google Cloud Storage (GCS)**:
+  - `credentials`: GCS credentials file content.
+  - `bucket`: GCS bucket name where the snapshot is stored.
+  - `base_path`: Path to repository data within the bucket.
+  - `snapshot_name`: Name of the snapshot to restore from.
+  - `compress`: (Optional) Whether to store metadata files in compressed format.
+  - `chunk_size`: (Optional) Size of chunks for large files during snapshot creation.
+
+- **Amazon S3 and S3-compatible services**:
+  - `region`: AWS/OCI region.
+  - `bucket`: S3/OCI bucket name where the snapshot is stored.
+  - `base_path`: Path to repository data within the bucket.
+  - `snapshot_name`: Name of the snapshot to restore from.
+  - `endpoint`: (Optional) The service endpoint for S3-compatible services.
+  - `server_side_encryption`: (Optional) Whether to enable server-side encryption.
+  - `compress`: (Optional) Whether to store metadata files in compressed format.
+  - `chunk_size`: (Optional) Size of chunks for large files during snapshot creation.
+
+- **Microsoft Azure**:
+  - `container`: Azure container name where the snapshot is stored.
+  - `base_path`: Path to repository data within the container.
+  - `snapshot_name`: Name of the snapshot to restore from.
+  - `compress`: (Optional) Whether to store metadata files in compressed format.
+  - `chunk_size`: (Optional) Size of chunks for large files during snapshot creation.
+
+## Register the snapshot repository
+
+Configure Aiven for OpenSearch to identify and access the location where your
+snapshot is stored.
+
+### Amazon S3 and S3-compatible services
+
+For Amazon S3 or any S3-compatible service, use one of the following methods to
+register the snapshot repository.
+
+<Tabs groupId="config-methods">
+<TabItem value="api" label="API" default>
+
+```bash
+curl -X PUT "https://SERVICE_NAME.aivencloud.com/_snapshot/my-snapshot-repo" \
+-H "Authorization: Bearer API_TOKEN" \
+-H "Content-Type: application/json" \
+-d '{
+  "type": "s3",
+  "settings": {
+    "bucket": "my-bucket",
+    "region": "us-west-2",
+    "endpoint": "https://s3.amazonaws.com"
+  }
+}'
+```
+
+**Parameters:**
+
+- `SERVICE_NAME`: The name of your Aiven OpenSearch service.
+- `API_TOKEN`: The [API token for authentication](/docs/platform/howto/create_authentication_token).
+- `bucket`: The S3 bucket name where the snapshot is stored.
+- `region`: The AWS region where the bucket is located.
+- `endpoint`: (Optional) The service endpoint for S3-compatible services.
+
+</TabItem>
+<TabItem value="cli" label="CLI">
+
+```bash
+avn service elasticsearch-register-repository \
```

Comment:
`avn service update -c s3_migration.region="asdsadasdasd" -c s3_migration.snapshot_name="dddd" -c s3_migration.base_path="/a/b/c/d" -c s3_migration.bucket="b" -c s3_migration.access_key="access_key" -c s3_migration.secret_key="secret_key_1" --project <p-name> <s_name>`

so `s3_migration.*` all properties mentioned above

Reply:
And same for gcs and azure the difference is in the suffux:
`azure_migration.*` and `gcs_migration.*`

Reply:
Is this the correct command to register a repository?